core/configuration.py

```diff
@@ -6,73 +6,91 @@
 
 
 logging.basicConfig(level=logging.DEBUG,
-					format='[ConfigurationManager] %(asctime)s %(levelname)s %(message)s')
+                    format='[ConfigurationManager] %(asctime)s %(levelname)s %(message)s')
 
 class ConfigurationManager(object):
-	"""
-	Configuration Manager
+    """
+    Configuration Manager
 
-	Singleton class that manages the configuration of all other modules.
+    Singleton class that manages the configuration of all other modules.
 
-	"""
+    """
 
-	def __init__(self):
-		"""
-		Virtually private constructor.
-		"""
-		self.question_format = "{question} [default = {default}] "
-		self._config = None
+    def __init__(self):
+        """
+        Virtually private constructor.
+        """
+        self.question_format = "{question} [default = {default}] "
+        self._config = None
 
-	def reset(self):
-		"""
-		Resets the configuration by deleting it. Bootstrap must be called so that
-		the class works as intended.
-		"""
-		self._config = None
-		return self
+    def reset(self):
+        """
+        Resets the configuration by deleting it. Bootstrap must be called so that
+        the class works as intended.
+        """
+        self._config = None
+        return self
 
-	def bootstrap(self, config_filepath="core/configuration.ini", input_function=None):
-		"""
-		Do nothing if config instance already exists. Otherwise, check if
-		core/configuration.ini exists. If not, call run_setup_mode. Then load
-		the config instance from configuration.ini
-		"""
-		if self._config:
-			return False
-		self.input_function = input_function if input_function else input
-		self.config_filepath = config_filepath
-		if not os.path.isfile(self.config_filepath):
-			self.run_setup_mode(self.config_filepath, self.input_function)
-		else:
-			self._config = configparser.ConfigParser()
-			self._config.read(self.config_filepath)
-		return True
+    def bootstrap(self, config_filepath="core/configuration.ini", input_function=None):
```

Comment:
So I noticed the tests don't test a `questions.csv` with more than one secret subsection (other than DBClient). Looking at the code it *seems* like this would work, but it would be better if our tests actually cover this.

We can make this work by passing an argument `questions_filepath` to `bootstrap()`, passing it all the way to `_run_setup_mode()`, and using a testing .csv with 2+ sections on the tests. (This also removes the magic string in the `_run_setup_mode()` method.)